src/lib.rs

```diff
@@ -53,3 +49,27 @@ impl Behaviour {
         }
     }
 }
+
+pub async fn load_secret_key_from_file(path: impl AsRef<Path>) -> Result<SecretKey> {
+    let bytes = fs::read(path).await?;
+    let secret_key = SecretKey::from_bytes(bytes)?;
+    Ok(secret_key)
+}
+
+#[derive(Debug, thiserror::Error)]
+#[error("Could not generate secret file at specified location. File {path} already exists")]
+pub struct FileAlreadyExistsAtSpecifiedLocation {
+    path: PathBuf,
+}
+
+pub async fn generate_secret_key_file(path: PathBuf) -> Result<SecretKey> {
+    if fs::metadata(&path).await?.is_file() {
+        return Err(anyhow::Error::from(FileAlreadyExistsAtSpecifiedLocation {
+            path: path.clone(),
+        }));
+    }
+    let keypair = Keypair::generate();
+    let secret_key = SecretKey::from(keypair);
+    let _ = fs::write(path, secret_key.clone()).await?;
```

Comment:
If you write first, you can avoid a clone because writing only requires a reference (I think).